skywalking/agent/__init__.py

```diff
@@ -306,6 +303,7 @@ def start(self) -> None:
             profile.init()
         if config.agent_meter_reporter_active:
             meter.init(force=True)  # force re-init after fork()
+        sampling.init(force=True)
```

Comment:
Can we check `sample_n_per_3_secs > 0` here and if not, do not even initialize the sampling service?

 ```suggestion
        if config.sample_n_per_3_secs > 0:
            sampling.init(force=True)
```

Same for the async variant below

Reply:
Sure